src/event_imp.rs

```diff
@@ -981,6 +986,450 @@ fn test_debug_ready() {
+    /// should be treated as a hint. For more details, see [readiness] in the
+    /// poll documentation.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::hup();
+    ///
+    /// assert!(interest.is_hup());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    /// [readiness]: ../struct.Poll.html#readiness-operations
+    #[inline]
+    #[cfg(unix)]
+    pub fn hup() -> Interests {
+        Interests(HUP)
+    }
+
+    /// Returns `Interests` representing error readiness.
+    ///
+    /// **Note that only readable and writable readiness is guaranteed to be
+    /// supported on all platforms**. This means that `error` readiness
+    /// should be treated as a hint. For more details, see [readiness] in the
+    /// poll documentation.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::error();
+    ///
+    /// assert!(interest.is_error());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    /// [readiness]: ../struct.Poll.html#readiness-operations
+    #[inline]
+    #[cfg(unix)]
+    pub fn error() -> Interests {
+        Interests(ERROR)
+    }
+
+    /// Returns `Interests` representing AIO completion readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::aio();
+    ///
+    /// assert!(interest.is_aio());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "ios",
+        target_os = "macos"
+    ))]
+    pub fn aio() -> Interests {
+        Interests(AIO)
+    }
+
+    /// Returns `Interests` representing LIO completion readiness.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::lio();
+    ///
+    /// assert!(interest.is_lio());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(target_os = "freebsd"))]
+    pub fn lio() -> Interests {
+        Interests(LIO)
+    }
+
+    /// Returns `Interests` representing priority (`EPOLLPRI`) readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::priority();
+    ///
+    /// assert!(interest.is_priority());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
+    pub fn priority() -> Interests {
```

Comment:
I believe priority is also registered by default.

Reply:
Roger